lib/Serialization/Deserialization.cpp

```diff
@@ -3057,8 +3057,11 @@ ModuleFile::getDeclCheckedImpl(DeclID DID, Optional<DeclContext *> ForcedContext
 
     // Decode the default argument kind.
     // FIXME: Default argument expression, if available.
-    if (auto defaultArg = getActualDefaultArgKind(rawDefaultArg))
+    if (auto defaultArg = getActualDefaultArgKind(rawDefaultArg)) {
       param->setDefaultArgumentKind(*defaultArg);
+      if (!blobData.empty())
+        param->setDefaultValueStringRepresentation(blobData);
```

Comment:
Probably I'm missing something. But, where the `blobData` come from? Isn't this `StringRef` backed by `SmallVector<uint64_t, 64> scratch;`? Is it safe to store it in AST nodes as a `StringRef`?

Reply:
Ah, I read the source of `BitstreamReader`, and I understand it now :)